File: src/test/fixtures/dart/lib/service.dart
import 'models.dart';

UserProfile createUser(String name, String email) {
  return UserProfile(
    id: 0,
    createdAt: DateTime.now(),
    updatedAt: DateTime.now(),
    name: name,
    email: email,
    age: 0,
  );
}

UserProfile getCompanyOwner(CompanyInfo company) {
  return company.getOwner();
}

void processEntity(BaseEntity entity) {
  print(entity.id);
}

TimestampedEntity getTimestamped(int id) {
  return TimestampedEntity(
    id: id,
    createdAt: DateTime.now(),
    updatedAt: DateTime.now(),
  );
}

File: src/test/fixtures/cpp/service.cpp
#include "models.h"
#include <iostream>

UserProfile createUser(const std::string& name, const std::string& email) {
    UserProfile user;
    user.name = name;
    user.email = email;
    user.age = 0;
    return user;
}

UserProfile getCompanyOwner(const CompanyInfo& company) {
    return company.getOwner();
}

void processEntity(const BaseEntity& entity) {
    std::cout << entity.id << std::endl;
}

TimestampedEntity getTimestamped(int id) {
    TimestampedEntity entity;
    entity.id = id;
    return entity;
}
